views.py

```diff
@@ -35,12 +40,38 @@ def get_context_data(self, **kwargs):
             'portal_id': self.hubspot_portal_id,
             'form_id': self.hubspot_form_id,
         }
+        kwargs['language_options'] = self.language_options()
         return super(BasePreloginView, self).get_context_data(**kwargs)
 
     @use_bootstrap3
     def dispatch(self, request, *args, **kwargs):
         return super(BasePreloginView, self).dispatch(request, *args, **kwargs)
 
+    def get(self, request, *args, **kwargs):
+        lang_prefix = kwargs.get('lang_code')
+        if lang_prefix and lang_prefix not in prelogin_lang_codes():
+            return HttpResponseRedirect(reverse(self.urlname))
+        self.activate_language_from_request(request, lang_prefix)
+        return super(BasePreloginView, self).get(request, *args, **kwargs)
+
+    def activate_language_from_request(self, request, lang_prefix):
+        if lang_prefix in prelogin_lang_codes():
+            lang = lang_prefix
+        else:
+            lang = translation.get_language_from_request(request)
+        request.session[translation.LANGUAGE_SESSION_KEY] = lang
+        request.session['django_language'] = lang
+        translation.activate(lang)
```

Comment:
Yes, that's right. Language setting is maintained (even when you go back to root url). I raised if this is desired behaviour in the spec. Looks like answer is no, so I am gonna remove that behaviour, (Line 62 is where that is taken care of)